### ISSUE TITLE
Body position gets offset when wrapping OffsetCenterOfMassShape in ScaledShape

### ISSUE THREAD

**Issue by mihe** [jrouwe/JoltPhysics#508]
I noticed something strange when trying to add support for scaling bodies.

Until now I've wrapped the body's individual shapes in a `ScaledShape` to represent the shape's scaling and then an `OffsetCenterOfMassShape` around the compound shape to represent the body's center-of-mass override, which has worked well, but now I tried to wrap that `OffsetCenterOfMassShape` in its own `ScaledShape` to represent the body's scale and all of a sudden the positions reported by `Body::GetPosition` doesn't line up with the body's actual position.

I hijacked `OffsetCenterOfMassShapeTest` to show you an example:

```cpp
void OffsetCenterOfMassShapeTest::Initialize() 
{
    CreateFloor();

    Ref sphere = new SphereShapeSettings(1.0f);
    Ref pre_scaled = new ScaledShape(sphere->Create().Get(), JPH::Vec3::sReplicate(1.25f));
    Ref with_com = new OffsetCenterOfMassShapeSettings(Vec3(0, 0, 5.0f), pre_scaled);
    Ref post_scaled = new ScaledShape(with_com->Create().Get(), JPH::Vec3::sReplicate(1.5f));

    Body &body_control = *mBodyInterface->CreateBody(BodyCreationSettings(with_com, RVec3(0, 5, 0), Quat::sIdentity(), EMotionType::Dynamic, Layers::MOVING));
    mBodyInterface->AddBody(body_control.GetID(), EActivation::Activate);

    Body &body_strange = *mBodyInterface->CreateBody(BodyCreationSettings(post_scaled, RVec3(5, 5, 0), Quat::sIdentity(), EMotionType::Dynamic, Layers::MOVING));
    mBodyInterface->AddBody(body_strange.GetID(), EActivation::Activate);

    Vec3 position = body_strange.GetPosition();

    // Prints "Position: (5.0, 5.0, 0.0)" despite the body being at (5.0, 5.0, 2.5)
    Trace("Position: (%.1f, %.1f, %.1f)", position.GetX(), position.GetY(), position.GetZ());
}
```

Here's what that looks like when looking at `(0, 0, 0)`, with +Z being to the left. The closer/yellow body reports being at Z-coordinate 0, but is clearly not.

![CenterOfMassOffset](https://user-images.githubusercontent.com/4884246/232206431-c7622d14-3b58-4350-a993-e206421e4671.png)

Am I doing something wrong here, or is this a bug?

**Comment by jrouwe** [comment]
There is a bug there indeed. The OffsetCOMShape has apparently never been used inside a ScaledShape. I fixed this in #509.

**Comment by mihe** [comment]
Thank you for the speedy fix! Much appreciated.
